src/store/config.ts

```diff
@@ -0,0 +1,1455 @@
+  ngeoSnappingTolerance: ngeoSnappingTolerance;
+  ngeoPointfilter: ngeoPointfilter;
+  ngeoCsvEncoding: ngeoCsvEncoding;
+  ngeoCsvExtension: ngeoCsvExtension;
+  ngeoCsvIncludeHeader: ngeoCsvIncludeHeader;
+  ngeoCsvQuote: ngeoCsvQuote;
+  ngeoCsvSeparator: ngeoCsvSeparator;
+  ngeoScaleSelectorOptions: ngeoScaleSelectorOptions;
+  ngeoProfileOptions: ngeoProfileOptions;
+  ngeoGeolocationOptions: ngeoGeolocationOptions;
+  ngeoStreetviewOptions: ngeoStreetviewOptions;
+  gmfTreeUrl: gmfTreeUrl;
+  gmfShortenerCreateUrl: gmfShortenerCreateUrl;
+  gmfRasterUrl: gmfRasterUrl;
+  gmfProfileJsonUrl: gmfProfileJsonUrl;
+  gmfLayersUrl: gmfLayersUrl;
+  gmfPrintUrl: gmfPrintUrl;
+  gmfOptions: gmfOptions;
+  gmfProjectionsOptions: gmfProjectionsOptions;
+  gmfTreeManagerModeFlush: gmfTreeManagerModeFlush;
+  gmfContextualDataOptions: gmfContextualDataOptions;
+  gmfObjectEditingToolsOptions: gmfObjectEditingToolsOptions;
+  gmfPermalinkOptions: gmfPermalinkOptions;
+  gmfPrintOptions: gmfPrintOptions;
+  gmfThemesOptions: gmfThemesOptions;
+  gmfFloors: gmfFloors;
+  gmfAuthenticationConfig: gmfAuthenticationConfig;
+  gmfExternalOGCServers: gmfExternalOGCServers;
+  gmfSearchGroups: gmfSearchGroups;
+  gmfAuthenticationNoReloadRole: gmfAuthenticationNoReloadRole;
+  gmfCsvFilename: gmfCsvFilename;
+  gmfBackgroundLayerSelectorOptions: gmfBackgroundLayerSelectorOptions;
+  gmfDisclaimerOptions: gmfDisclaimerOptions;
+  gmfQueryGrid: gmfQueryGrid;
+  gmfDisplayQueryGridOptions: gmfDisplayQueryGridOptions;
+  gmfDisplayQueryWindowOptions: gmfDisplayQueryWindowOptions;
+  gmfElevationOptions: gmfElevationOptions;
+  gmfProfileOptions: gmfProfileOptions;
+  gmfSearchOptions: gmfSearchOptions;
+  gmfMousePositionOptions: gmfMousePositionOptions;
+  gmfMobileMeasureAreaOptions: gmfMobileMeasureAreaOptions;
+  gmfMobileMeasureLengthOptions: gmfMobileMeasureLengthOptions;
+  gmfMobileMeasurePointOptions: gmfMobileMeasurePointOptions;
+  gmfLayerTreeOptions: gmfLayerTreeOptions;
+  gmfShareOptions: gmfShareOptions;
+  gmfEditFeatureOptions: gmfEditFeatureOptions;
+  gmfCustomJavascriptUrl: gmfCustomJavascriptUrl;
+  gmfCustomStylesheetUrl: gmfCustomStylesheetUrl;
+  gmfTwoFactorAuth: gmfTwoFactorAuth;
+};
+
+export type APIConfig = {
+  themesUrl?: string;
+  localeUrl?: string;
+  searchUrl?: string;
+  projection?: string;
+  projections?: {
+    [x: string]: Projection;
+  };
+  resolutions?: number[];
+  extent?: number[];
+  backgroundLayer?: string;
+  queryableLayers?: string[];
+};
+
+export class ConfigModel {
+  /**
+   * The observable user's properties. The default user is empty.
+   * @private
+   */
+  configuration_: BehaviorSubject<Configuration>;
+
+  /**
+   * @return the observable user's properties.
+   */
+  getConfig(): BehaviorSubject<Configuration> {
+    return this.configuration_;
+  }
+
+  setConfig(config: Configuration) {
+    this.configuration_ = new BehaviorSubject<Configuration>(config);
```

Comment:
I now it's too soon to review, but this will not work. You probably want this line the constructor, and here:
`this.configuration_.next(config)`